Userland/Libraries/LibRequests/NetworkErrorEnum.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2024, the Ladybird developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+namespace Requests {
+
+enum class NetworkError {
+    UnableToResolveProxy,
+    UnableToResolveHost,
+    UnableToConnect,
+    TimeoutReached,
+    TooManyRedirects,
```

Comment:
I don't think this can ever happen? RS doesn't follow redirects (ResourceLoader does, which starts another request).
though iirc @ADKaster wanted to make curl follow redirects at some point? [imo a bad idea]

Either way, it's okay to have this.

Reply:
> though iirc @ADKaster wanted to make curl follow redirects at some point? 

This was mostly a debugging thing when looking at a local python3 https server

Reply:
Ah gotcha, makes sense